src/main/java/io/vertx/core/http/impl/Http2ServerResponse.java

```diff
@@ -453,6 +453,8 @@ void write(ByteBuf chunk, boolean end, Handler<AsyncResult<Void>> handler) {
       boolean sent = checkSendHeaders(end && !hasBody && trailers == null);
       if (hasBody || (!sent && end)) {
         stream.writeData(chunk, end && trailers == null, handler);
+      } else {
+        handler.handle(Future.succeededFuture());
```

Comment:
the handler should be called outside of the synchronized block like the endHandler and the bodyEndHandler